tests/unit/common/test_query_constructor.py

```diff
@@ -451,3 +451,52 @@ def test_filter_conversion(self):
 ;
 """)
         self.assertEqual(queries["expression_query"], expected_exp_query)
+
+
+class TestDetailQuery(unittest.TestCase):
+
+    def test_cell_numeric(self):
+        query = query_constructor.create_field_detail_query(
+            "cell.genes_detected",
+            "cell",
+            "cellkey",
+            "numeric")
+
+        expected_sql = """
+SELECT MIN(cell.genes_detected), MAX(cell.genes_detected)
+FROM cell 
+;
+"""  # noqa: W291
+        self.assertEqual(query, expected_sql)
+
+    def test_cell_categorical(self):
+
+        query = query_constructor.create_field_detail_query(
+            "cell.cell_suspension_id",
+            "cell",
+            "cellkey",
+            "categorical")
+
+        expected_sql = """
+SELECT cell.cell_suspension_id, COUNT(cell.cellkey)
+FROM cell 
+GROUP BY cell.cell_suspension_id
+;
+"""  # noqa: W291
```

Comment:
nit: can't you just remove the whitespace?

https://lintlyci.github.io/Flake8Rules/rules/W291.html

Reply:
The template has a `FROM cell {join}\n`, so if `join` is empty, there's a space at the end of the line. I could make it `FROM cell{join}\n` and put a whitespace in the beginning of the string passed to `join`, but I thought that looked weirder.

Reply:
Oh nevermind me then. Thought the violation was on line 485